hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,249 @@
+
+import com.google.protobuf.ByteString;
+import com.hedera.node.app.service.evm.utils.EthSigsUtils;
+import com.hedera.node.app.service.mono.config.AccountNumbers;
+import com.hedera.node.app.service.mono.context.annotations.CompositeProps;
+import com.hedera.node.app.service.mono.context.properties.PropertySource;
+import com.hedera.node.app.service.mono.ledger.accounts.AliasManager;
+import com.hedera.node.app.service.mono.ledger.accounts.HederaAccountCustomizer;
+import com.hedera.node.app.service.mono.ledger.backing.BackingStore;
+import com.hedera.node.app.service.mono.ledger.ids.EntityIdSource;
+import com.hedera.node.app.service.mono.legacy.core.jproto.JEd25519Key;
+import com.hedera.node.app.service.mono.legacy.core.jproto.JKey;
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.KeyList;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.asn1.sec.SECNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECPublicKeyParameters;
+import org.bouncycastle.util.encoders.DecoderException;
+import org.bouncycastle.util.encoders.Hex;
+
+/**
+ * Encapsulates the logic for reading blocked accounts from file and creating them.
+ */
+@Singleton
+public class BlocklistAccountCreator {
+    private static final Logger log = LogManager.getLogger(BlocklistAccountCreator.class);
+    private final String blocklistFileName;
+    private final Supplier<HederaAccount> accountSupplier;
+    private final EntityIdSource ids;
+    private final BackingStore<AccountID, HederaAccount> accounts;
+    private final Supplier<JEd25519Key> genesisKeySource;
+    private final PropertySource properties;
+    private final AliasManager aliasManager;
+    private JKey genesisKey;
+    private final List<HederaAccount> accountsCreated = new ArrayList<>();
+    private AccountNumbers accountNumbers;
+
+    /**
+     * @param evmAddress the EVM address of the blocked account
+     * @param memo      the memo of the blocked account
+     */
+    record BlockedInfo(ByteString evmAddress, String memo) {}
+
+    @Inject
+    public BlocklistAccountCreator(
+            final Supplier<HederaAccount> accountSupplier,
+            final EntityIdSource ids,
+            final BackingStore<AccountID, HederaAccount> accounts,
+            final Supplier<JEd25519Key> genesisKeySource,
```

Comment:
(this is mysterious to me: how the heck does the thing that does the injecting know to inject the "genesis key source" here?  is there only _one_ `Supplier<JEd25519Key>` _ever_ injected?  Or what?  And where is it specified anyway?)